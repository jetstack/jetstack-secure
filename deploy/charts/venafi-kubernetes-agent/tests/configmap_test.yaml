suite: test the contents of the config.yaml
templates:
  - configmap.yaml
  - NOTES.txt
release:
  name: test
  namespace: test-ns
tests:
  - it: defaults
    asserts:
      - matchSnapshot: {}

  - it: custom-period
    set:
      config.period: 1m
    asserts:
      - matchSnapshot: {}

  - it: custom-cluster-name
    set:
      config.clusterName: "cluster-1 region-1 cloud-1 "
    asserts:
      - matchSnapshot: {}

  - it: custom-cluster-description
    set:
      config.clusterDescription: |
        A cloud hosted Kubernetes cluster hosting production workloads.

        team: team-1
        email: team-1@example.com
        purpose: Production workloads
    asserts:
      - matchSnapshot: {}

  - it: custom-configmap
    set:
      config:
        configmap:
          name: agent-custom-config
    asserts:
      - matchSnapshotRaw: {}
