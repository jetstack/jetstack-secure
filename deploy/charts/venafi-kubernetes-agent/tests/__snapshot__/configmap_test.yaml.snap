custom-cluster-description:
  1: |
    raw: |2
      - Check the credentials Secret exists: "agent-credentials"
      > kubectl get secret -n test-ns agent-credentials
      - Check the application is running:
      > kubectl get pods -n test-ns -l app.kubernetes.io/instance=test

      - Check the application logs for successful connection to the platform:
      > kubectl logs -n test-ns -l app.kubernetes.io/instance=test
  2: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: ""
        cluster_description: "A cloud hosted Kubernetes cluster hosting production workloads.\n\nteam: team-1\nemail: team-1@example.com\npurpose: Production workloads\n"
        server: "https://api.venafi.cloud/"
        period: "0h1m0s"
        venafi-cloud:
          uploader_id: "no"
          upload_path: "/v1/tlspk/upload/clusterdata"
        data-gatherers:
        # gather k8s apiserver version information
        - kind: "k8s-discovery"
          name: "k8s-discovery"
        # pods data is used in the pods and application_versions packages
        - kind: "k8s-dynamic"
          name: "k8s/pods"
          config:
            resource-type:
              resource: pods
              version: v1
        - kind: "k8s-dynamic"
          name: "k8s/namespaces"
          config:
            resource-type:
              resource: namespaces
              version: v1
        # gather services for pod readiness probe rules
        - kind: "k8s-dynamic"
          name: "k8s/services"
          config:
            resource-type:
              resource: services
              version: v1
        # gather higher level resources to ensure data to determine ownership is present
        - kind: "k8s-dynamic"
          name: "k8s/deployments"
          config:
            resource-type:
              version: v1
              resource: deployments
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/statefulsets"
          config:
            resource-type:
              version: v1
              resource: statefulsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/daemonsets"
          config:
            resource-type:
              version: v1
              resource: daemonsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/jobs"
          config:
            resource-type:
              version: v1
              resource: jobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/cronjobs"
          config:
            resource-type:
              version: v1
              resource: cronjobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/ingresses"
          config:
            resource-type:
              group: networking.k8s.io
              version: v1
              resource: ingresses
        - kind: "k8s-dynamic"
          name: "k8s/secrets"
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/service-account-token
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=kubernetes.io/basic-auth
            - type!=kubernetes.io/ssh-auth
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: "k8s-dynamic"
          name: "k8s/certificates"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificates
        - kind: "k8s-dynamic"
          name: "k8s/certificaterequests"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificaterequests
        - kind: "k8s-dynamic"
          name: "k8s/issuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/clusterissuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasclusterissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaissuer"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaclusterissuers"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/mutatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: mutatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/validatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: validatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/gateways"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: gateways
        - kind: "k8s-dynamic"
          name: "k8s/virtualservices"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: virtualservices
        - kind: "k8s-dynamic"
          name: "k8s/routes"
          config:
            resource-type:
              version: v1
              group: route.openshift.io
              resource: routes
        - kind: "k8s-dynamic"
          name: "k8s/venaficonnections"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficonnections
        - kind: "k8s-dynamic"
          name: "k8s/venaficlusterissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficlusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/venafiissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venafiissuers
        - kind: "k8s-dynamic"
          name: "k8s/fireflyissuers"
          config:
            resource-type:
              group: firefly.venafi.com
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/stepissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepissuers
        - kind: "k8s-dynamic"
          name: "k8s/stepclusterissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/originissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: originissuers
        - kind: "k8s-dynamic"
          name: "k8s/clusteroriginissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: clusteroriginissuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaclusterissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaclusterissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: clusterissuers
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: venafi-kubernetes-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: venafi-kubernetes-agent-0.0.0
      name: agent-config
      namespace: test-ns
custom-cluster-name:
  1: |
    raw: |2
      - Check the credentials Secret exists: "agent-credentials"
      > kubectl get secret -n test-ns agent-credentials
      - Check the application is running:
      > kubectl get pods -n test-ns -l app.kubernetes.io/instance=test

      - Check the application logs for successful connection to the platform:
      > kubectl logs -n test-ns -l app.kubernetes.io/instance=test
  2: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: "cluster-1 region-1 cloud-1 "
        cluster_description: ""
        server: "https://api.venafi.cloud/"
        period: "0h1m0s"
        venafi-cloud:
          uploader_id: "no"
          upload_path: "/v1/tlspk/upload/clusterdata"
        data-gatherers:
        # gather k8s apiserver version information
        - kind: "k8s-discovery"
          name: "k8s-discovery"
        # pods data is used in the pods and application_versions packages
        - kind: "k8s-dynamic"
          name: "k8s/pods"
          config:
            resource-type:
              resource: pods
              version: v1
        - kind: "k8s-dynamic"
          name: "k8s/namespaces"
          config:
            resource-type:
              resource: namespaces
              version: v1
        # gather services for pod readiness probe rules
        - kind: "k8s-dynamic"
          name: "k8s/services"
          config:
            resource-type:
              resource: services
              version: v1
        # gather higher level resources to ensure data to determine ownership is present
        - kind: "k8s-dynamic"
          name: "k8s/deployments"
          config:
            resource-type:
              version: v1
              resource: deployments
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/statefulsets"
          config:
            resource-type:
              version: v1
              resource: statefulsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/daemonsets"
          config:
            resource-type:
              version: v1
              resource: daemonsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/jobs"
          config:
            resource-type:
              version: v1
              resource: jobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/cronjobs"
          config:
            resource-type:
              version: v1
              resource: cronjobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/ingresses"
          config:
            resource-type:
              group: networking.k8s.io
              version: v1
              resource: ingresses
        - kind: "k8s-dynamic"
          name: "k8s/secrets"
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/service-account-token
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=kubernetes.io/basic-auth
            - type!=kubernetes.io/ssh-auth
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: "k8s-dynamic"
          name: "k8s/certificates"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificates
        - kind: "k8s-dynamic"
          name: "k8s/certificaterequests"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificaterequests
        - kind: "k8s-dynamic"
          name: "k8s/issuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/clusterissuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasclusterissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaissuer"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaclusterissuers"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/mutatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: mutatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/validatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: validatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/gateways"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: gateways
        - kind: "k8s-dynamic"
          name: "k8s/virtualservices"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: virtualservices
        - kind: "k8s-dynamic"
          name: "k8s/routes"
          config:
            resource-type:
              version: v1
              group: route.openshift.io
              resource: routes
        - kind: "k8s-dynamic"
          name: "k8s/venaficonnections"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficonnections
        - kind: "k8s-dynamic"
          name: "k8s/venaficlusterissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficlusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/venafiissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venafiissuers
        - kind: "k8s-dynamic"
          name: "k8s/fireflyissuers"
          config:
            resource-type:
              group: firefly.venafi.com
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/stepissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepissuers
        - kind: "k8s-dynamic"
          name: "k8s/stepclusterissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/originissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: originissuers
        - kind: "k8s-dynamic"
          name: "k8s/clusteroriginissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: clusteroriginissuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaclusterissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaclusterissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: clusterissuers
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: venafi-kubernetes-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: venafi-kubernetes-agent-0.0.0
      name: agent-config
      namespace: test-ns
custom-configmap:
  1: |
    |2
      You are using a custom configuration in the following ConfigMap: "agent-custom-config".

      DEPRECATION: The `cluster_id` configuration field is deprecated.
      If your configuration contains `cluster_id`, it will continue to work as a
      fallback, but please migrate to `cluster_name` to avoid ambiguity.
      - Check the credentials Secret exists: "agent-credentials"
      > kubectl get secret -n test-ns agent-credentials
      - Check the application is running:
      > kubectl get pods -n test-ns -l app.kubernetes.io/instance=test

      - Check the application logs for successful connection to the platform:
      > kubectl logs -n test-ns -l app.kubernetes.io/instance=test
custom-period:
  1: |
    raw: |2
      - Check the credentials Secret exists: "agent-credentials"
      > kubectl get secret -n test-ns agent-credentials
      - Check the application is running:
      > kubectl get pods -n test-ns -l app.kubernetes.io/instance=test

      - Check the application logs for successful connection to the platform:
      > kubectl logs -n test-ns -l app.kubernetes.io/instance=test
  2: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: ""
        cluster_description: ""
        server: "https://api.venafi.cloud/"
        period: "1m"
        venafi-cloud:
          uploader_id: "no"
          upload_path: "/v1/tlspk/upload/clusterdata"
        data-gatherers:
        # gather k8s apiserver version information
        - kind: "k8s-discovery"
          name: "k8s-discovery"
        # pods data is used in the pods and application_versions packages
        - kind: "k8s-dynamic"
          name: "k8s/pods"
          config:
            resource-type:
              resource: pods
              version: v1
        - kind: "k8s-dynamic"
          name: "k8s/namespaces"
          config:
            resource-type:
              resource: namespaces
              version: v1
        # gather services for pod readiness probe rules
        - kind: "k8s-dynamic"
          name: "k8s/services"
          config:
            resource-type:
              resource: services
              version: v1
        # gather higher level resources to ensure data to determine ownership is present
        - kind: "k8s-dynamic"
          name: "k8s/deployments"
          config:
            resource-type:
              version: v1
              resource: deployments
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/statefulsets"
          config:
            resource-type:
              version: v1
              resource: statefulsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/daemonsets"
          config:
            resource-type:
              version: v1
              resource: daemonsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/jobs"
          config:
            resource-type:
              version: v1
              resource: jobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/cronjobs"
          config:
            resource-type:
              version: v1
              resource: cronjobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/ingresses"
          config:
            resource-type:
              group: networking.k8s.io
              version: v1
              resource: ingresses
        - kind: "k8s-dynamic"
          name: "k8s/secrets"
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/service-account-token
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=kubernetes.io/basic-auth
            - type!=kubernetes.io/ssh-auth
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: "k8s-dynamic"
          name: "k8s/certificates"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificates
        - kind: "k8s-dynamic"
          name: "k8s/certificaterequests"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificaterequests
        - kind: "k8s-dynamic"
          name: "k8s/issuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/clusterissuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasclusterissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaissuer"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaclusterissuers"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/mutatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: mutatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/validatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: validatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/gateways"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: gateways
        - kind: "k8s-dynamic"
          name: "k8s/virtualservices"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: virtualservices
        - kind: "k8s-dynamic"
          name: "k8s/routes"
          config:
            resource-type:
              version: v1
              group: route.openshift.io
              resource: routes
        - kind: "k8s-dynamic"
          name: "k8s/venaficonnections"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficonnections
        - kind: "k8s-dynamic"
          name: "k8s/venaficlusterissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficlusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/venafiissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venafiissuers
        - kind: "k8s-dynamic"
          name: "k8s/fireflyissuers"
          config:
            resource-type:
              group: firefly.venafi.com
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/stepissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepissuers
        - kind: "k8s-dynamic"
          name: "k8s/stepclusterissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/originissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: originissuers
        - kind: "k8s-dynamic"
          name: "k8s/clusteroriginissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: clusteroriginissuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaclusterissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaclusterissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: clusterissuers
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: venafi-kubernetes-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: venafi-kubernetes-agent-0.0.0
      name: agent-config
      namespace: test-ns
defaults:
  1: |
    raw: |2
      - Check the credentials Secret exists: "agent-credentials"
      > kubectl get secret -n test-ns agent-credentials
      - Check the application is running:
      > kubectl get pods -n test-ns -l app.kubernetes.io/instance=test

      - Check the application logs for successful connection to the platform:
      > kubectl logs -n test-ns -l app.kubernetes.io/instance=test
  2: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: ""
        cluster_description: ""
        server: "https://api.venafi.cloud/"
        period: "0h1m0s"
        venafi-cloud:
          uploader_id: "no"
          upload_path: "/v1/tlspk/upload/clusterdata"
        data-gatherers:
        # gather k8s apiserver version information
        - kind: "k8s-discovery"
          name: "k8s-discovery"
        # pods data is used in the pods and application_versions packages
        - kind: "k8s-dynamic"
          name: "k8s/pods"
          config:
            resource-type:
              resource: pods
              version: v1
        - kind: "k8s-dynamic"
          name: "k8s/namespaces"
          config:
            resource-type:
              resource: namespaces
              version: v1
        # gather services for pod readiness probe rules
        - kind: "k8s-dynamic"
          name: "k8s/services"
          config:
            resource-type:
              resource: services
              version: v1
        # gather higher level resources to ensure data to determine ownership is present
        - kind: "k8s-dynamic"
          name: "k8s/deployments"
          config:
            resource-type:
              version: v1
              resource: deployments
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/statefulsets"
          config:
            resource-type:
              version: v1
              resource: statefulsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/daemonsets"
          config:
            resource-type:
              version: v1
              resource: daemonsets
              group: apps
        - kind: "k8s-dynamic"
          name: "k8s/jobs"
          config:
            resource-type:
              version: v1
              resource: jobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/cronjobs"
          config:
            resource-type:
              version: v1
              resource: cronjobs
              group: batch
        - kind: "k8s-dynamic"
          name: "k8s/ingresses"
          config:
            resource-type:
              group: networking.k8s.io
              version: v1
              resource: ingresses
        - kind: "k8s-dynamic"
          name: "k8s/secrets"
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/service-account-token
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=kubernetes.io/basic-auth
            - type!=kubernetes.io/ssh-auth
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: "k8s-dynamic"
          name: "k8s/certificates"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificates
        - kind: "k8s-dynamic"
          name: "k8s/certificaterequests"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: certificaterequests
        - kind: "k8s-dynamic"
          name: "k8s/issuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/clusterissuers"
          config:
            resource-type:
              group: cert-manager.io
              version: v1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasissuers
        - kind: "k8s-dynamic"
          name: "k8s/googlecasclusterissuers"
          config:
            resource-type:
              group: cas-issuer.jetstack.io
              version: v1beta1
              resource: googlecasclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaissuer"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaissuers
        - kind: "k8s-dynamic"
          name: "k8s/awspcaclusterissuers"
          config:
            resource-type:
              group: awspca.cert-manager.io
              version: v1beta1
              resource: awspcaclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/mutatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: mutatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/validatingwebhookconfigurations"
          config:
            resource-type:
              group: admissionregistration.k8s.io
              version: v1
              resource: validatingwebhookconfigurations
        - kind: "k8s-dynamic"
          name: "k8s/gateways"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: gateways
        - kind: "k8s-dynamic"
          name: "k8s/virtualservices"
          config:
            resource-type:
              group: networking.istio.io
              version: v1alpha3
              resource: virtualservices
        - kind: "k8s-dynamic"
          name: "k8s/routes"
          config:
            resource-type:
              version: v1
              group: route.openshift.io
              resource: routes
        - kind: "k8s-dynamic"
          name: "k8s/venaficonnections"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficonnections
        - kind: "k8s-dynamic"
          name: "k8s/venaficlusterissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venaficlusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/venafiissuers"
          config:
            resource-type:
              group: jetstack.io
              version: v1alpha1
              resource: venafiissuers
        - kind: "k8s-dynamic"
          name: "k8s/fireflyissuers"
          config:
            resource-type:
              group: firefly.venafi.com
              version: v1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/stepissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepissuers
        - kind: "k8s-dynamic"
          name: "k8s/stepclusterissuers"
          config:
            resource-type:
              group: certmanager.step.sm
              version: v1beta1
              resource: stepclusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/originissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: originissuers
        - kind: "k8s-dynamic"
          name: "k8s/clusteroriginissuers"
          config:
            resource-type:
              group: cert-manager.k8s.cloudflare.com
              version: v1
              resource: clusteroriginissuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/freeipaclusterissuers"
          config:
            resource-type:
              group: certmanager.freeipa.org
              version: v1beta1
              resource: clusterissuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: issuers
        - kind: "k8s-dynamic"
          name: "k8s/ejbcaclusterissuers"
          config:
            resource-type:
              group: ejbca-issuer.keyfactor.com
              version: v1alpha1
              resource: clusterissuers
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: venafi-kubernetes-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: venafi-kubernetes-agent-0.0.0
      name: agent-config
      namespace: test-ns
